examples/allocator-service/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"agones.dev/agones/pkg/apis/stable/v1alpha1"
+	"agones.dev/agones/pkg/client/clientset/versioned"
+	"agones.dev/agones/pkg/util/runtime" // for the logger
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+const namespace = "default"
+const fleetname = "simple-udp"
+const generatename = "simple-udp-"
+
+var (
+	logger  = runtime.NewLoggerWithSource("main")
+	address = "NoReadyGS" // default response if no gameservers are available
+	port    int32         // 0 until populated
+)
+
+// A handler for the web server
+type handler func(w http.ResponseWriter, r *http.Request)
+
+// The structure of the json response
+type result struct {
+	IPAddress string `json:"address"`
+	Port      int32  `json:"port"`
+}
+
+// Move a replica from ready to allocated and return the ip and port
+func getIPAddress() (string, int32) {
+	// Create the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.WithError(err).Fatal("Could not create in cluster config")
+	}
+
+	// Access to the Agones resources through the Agones Clientset
+	agonesClient, err := versioned.NewForConfig(config)
```

Comment:
Ouch.  I hadn't thought of that.

So I need to create it in main, but after the user has been authenticated?  I'll give that a try.

Reply:
I'm not sure I implemented the solution to this the right way.  What I'm hoping I did is create the client once and then point to it when using it later.

Reply:
It's not perfect -- i'd probably create a struct, and have it as a data member, and then create an instance of that in main.

I'm 97% sure you could use the functions from the object to be `HandleFunc()` as well.

That being said - it will probably do for this example.